py/helpers/cache/cache_helper.py

```diff
@@ -0,0 +1,34 @@
+import hashlib
+import json
+
+
+def sort_nested_dict(d):
+  if isinstance(d, dict):
+    return {k: sort_nested_dict(v) for k, v in sorted(d.items())}
+  elif isinstance(d, list):
+    return [sort_nested_dict(x) for x in d]
+  else:
+    return d
+
+class CacheKeyMaker:
+
+  def __init__(self):
+    pass
+
+  def _serialize(self, *args, **kwargs):
+    """Serialize arguments to create a unique key"""
+    combined = list(sorted(args)) + list(sort_nested_dict(kwargs.values()))
+    serialized_combined = json.dumps(sort_nested_dict(combined), sort_keys=True)
+    return serialized_combined
+
+  def make_key(self, *args, **kwargs):
+    """Generate a cache key based on provided arguments"""
+    serialized = self._serialize(*args, **kwargs)
+    key = f"{hashlib.md5(serialized.encode()).hexdigest()}"
+    return key
+
+  def make_custom_key(self, custom_part, *args, **kwargs):
```

Comment:
Related to my question about colons below. What are custom keys and where would they be used in Sight?

Reply:
so , the custom part is coloned part of the key added with md5 hash of long string , eg 
custom part = `user:12345:settings`
custom_key = `{cusotm_part}:{setting_id}`

so ,`make_custom_key` function makes the custom readable cache_key and not the gibberish hash key